### PR TITLE
Controllerspec

### PR DESCRIPTION
**Pull request by TaiwoEnoch**
## I create this PR to feature controllers specs requirement as listed below ⬇️ 

# Project requirements

- [ ] Add RSpec as a gem to your project.
- [ ] Create a Request spec file for all your controllers.
- [ ] Make sure that for each action you check:
     - [ ] If response status was correct.
     - [ ] If a correct template was rendered.
     - [ ] If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      expect(response.body).to include('<h1>Posts of all users will display here</h1>')
+    end
+  end
+
+  describe 'GET #show' do
+    before { get '/users/1' }
+
+    it 'returns a successful response' do
+      expect(response).to be_successful
+    end
```
Comment (franclobo) [inline spec/requests/posts_controller_spec.rb]: - [ ] This test returns an error, please 🙏🏽 correct it to pass all the tests 👇🏽 

```bash
2) UsersController GET #show returns a successful response
     Failure/Error: expect(response).to be_successful
       expected `#<ActionDispatch::TestResponse:0x00007f217edf51f0 @mon_data=#<Monitor:0x00007f217dc35de8>, @mon_data_...ache=>true}, @request=#<ActionDispatch::Request GET "http://www.example.com/users/1" for 127.0.0.1>>.successful?` to be truthy, got false
     # ./spec/requests/posts_controller_spec.rb:24:in `block (3 levels) in <main>'
```

File: spec/requests/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    it 'renders the show template' do
+      expect(response).to render_template(:show)
+    end
```
Comment (franclobo) [inline spec/requests/posts_controller_spec.rb]: - [ ] This test returns an error, please 🙏🏽 correct it to pass all the tests 👇🏽 

```bash
3) UsersController GET #show renders the show template
     Failure/Error: expect(response).to render_template(:show)
       expecting <"show"> but rendering with <[]>
     # ./spec/requests/posts_controller_spec.rb:28:in `block (3 levels) in <main>'
```

File: spec/requests/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    it 'includes correct placeholder text in the response body' do
+      expect(response.body).to include('<h1>Posts of all users will display here</h1>')
+    end
```
Comment (franclobo) [inline spec/requests/posts_controller_spec.rb]: - [ ] This test returns an error, please 🙏🏽 correct it to pass all the tests 👇🏽 

```bash
 4) UsersController GET #show includes correct placeholder text in the response body
     Failure/Error: expect(response.body).to include('<h1>Posts of all users will display here</h1>')
       expected "User not found" to include "<h1>Posts of all users will display here</h1>"
     # ./spec/requests/posts_controller_spec.rb:32:in `block (3 levels) in <main>'
```

File: spec/requests/posts_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      expect(response).to be_successful
+    end
+
+    it 'renders the index template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'includes correct placeholder text in the response body' do
+      expect(response.body).to include('<h1>Posts of all users will display here</h1>')
+    end
```
Comment (franclobo) [inline spec/requests/posts_controller_spec.rb]: - [ ] This test returns an error, please 🙏🏽 correct it to pass all the tests 👇🏽 

```bash
1) UsersController GET #index includes correct placeholder text in the response body
     Failure/Error: expect(response.body).to include('<h1>Posts of all users will display here</h1>')
     
       expected "<!DOCTYPE html>\n<html>\n  <head>\n    <title>Socialmedialapp</title>\n    <meta name=\"viewport\" c...  </head>\n\n  <body>\n    <h1>List of all the users will display here</h1>\n\n  </body>\n</html>\n" to include "<h1>Posts of all users will display here</h1>"
       Diff:
       @@ -1,33 +1,65 @@
       -<h1>Posts of all users will display here</h1>
       +<!DOCTYPE html>
       +<html>
       +  <head>
       +    <title>Socialmedialapp</title>
       +    <meta name="viewport" content="width=device-width,initial-scale=1">
       +    
       +    
       +
       +    <link rel="stylesheet" href="/assets/application-e0cf9d8fcb18bf7f909d8d91a5e78499f82ac29523d475bf3a9ab265d5e2b451.css" data-turbo-track="reload" />
       +    <script type="importmap" data-turbo-track="reload">{
       +  "imports": {
       +    "application": "/assets/application-37f365cbecf1fa2810a8303f4b6571676fa1f9c56c248528bc14ddb857531b95.js",
       +    "@hotwired/turbo-rails": "/assets/turbo.min-f309baafa3ae5ad6ccee3e7362118b87678d792db8e8ab466c4fa284dd3a4700.js",
       +    "@hotwired/stimulus": "/assets/stimulus.min-d03cf1dff41d6c5698ec2c5d6a501615a7a33754dbeef8d1edd31c928d17c652.js",
       +    "@hotwired/stimulus-loading": "/assets/stimulus-loading-1fc59770fb1654500044afd3f5f6d7d00800e5be36746d55b94a2963a7a228aa.js",
       +    "controllers/application": "/assets/controllers/application-368d98631bccbf2349e0d4f8269afb3fe9625118341966de054759d96ea86c7e.js",
       +    "controllers/hello_controller": "/assets/controllers/hello_controller-549135e8e7c683a538c3d6d517339ba470fcfb79d62f738a0a089ba41851a554.js",
       +    "controllers": "/assets/controllers/index-2db729dddcc5b979110e98de4b6720f83f91a123172e87281d5a58410fc43806.js"
       +  }
       +}</script>
       +<link rel="modulepreload" href="/assets/application-37f365cbecf1fa2810a8303f4b6571676fa1f9c56c248528bc14ddb857531b95.js">
       +<link rel="modulepreload" href="/assets/turbo.min-f309baafa3ae5ad6ccee3e7362118b87678d792db8e8ab466c4fa284dd3a4700.js">
       +<link rel="modulepreload" href="/assets/stimulus.min-d03cf1dff41d6c5698ec2c5d6a501615a7a33754dbeef8d1edd31c928d17c652.js">
       +<link rel="modulepreload" href="/assets/stimulus-loading-1fc59770fb1654500044afd3f5f6d7d00800e5be36746d55b94a2963a7a228aa.js">
       +<script src="/assets/es-module-shims.min-4ca9b3dd5e434131e3bb4b0c1d7dff3bfd4035672a5086deec6f73979a49be73.js" async="async" data-turbo-track="reload"></script>
       +<script type="module">import "application"</script>
       +  </head>
       +
       +  <body>
       +    <h1>List of all the users will display here</h1>
       +
       +  </body>
       +</html>
       
     # ./spec/requests/posts_controller_spec.rb:16:in `block (3 levels) in <main>'
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @TaiwoEnoch, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- All user:request:spec tests passed ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @TaiwoEnoch , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented changes requested to ensure all tests are passing 
✔️  Added all tests scenarios for the users and posts controllers

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**